python-package/xgboost/spark/core.py

```diff
@@ -1027,25 +1042,52 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
                 else:
                     base_margin = None
 
+                data = {}
                 preds = model.predict(
                     X,
                     base_margin=base_margin,
                     validate_features=False,
                     **predict_params,
                 )
-                yield pd.Series(preds)
+                data["prediction"] = pd.Series(preds)
+
+                if pred_contrib_col_name:
+                    contribs = pred_contribs(model, X, base_margin)
+                    assert len(contribs.shape) == 2
```

Comment:
This is not necessarily true. See doc/prediction.rst.

Reply:
Done.